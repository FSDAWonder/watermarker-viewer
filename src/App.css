
.App {
  text-align: center;
}

.iframe{
  margin: 5px;
  border:none;
  height: 500px;
  width: 80%;
}