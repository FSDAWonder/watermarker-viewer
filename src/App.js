import React, { useState } from "react";
import DocViewer, { DocViewerRenderers } from "@cyntler/react-doc-viewer";
import "./App.css";
import WaterMarker from "./Components/WaterMarker";

const App = () => {
  const [pdfUrl, setPdfUrl] = useState("");

  const docs = [
    { uri: require("./sample/xpro.pdf"), fileName: "pdf", fileType: "pdf" },
    // { uri: "https://docs.google.com/spreadsheets/d/1ylrv10lHZZRRqdLvo_5RTKFwYA8jsOtI_TkXGJxzeWI/edit#gid=0"},

    // { uri: require("./sample/xpro.docx"), fileName: "doc1", fileType: "docx" },
  ];

  return (
    <div className="App">
      <h1 className="heading">Document Viewer</h1>
      <WaterMarker docs={docs[0].uri} setPdfUrl={setPdfUrl} />
     
      {pdfUrl ? (
        <iframe className="iframe" src={pdfUrl} title="description">
         
        </iframe>
      ) : (
        <DocViewer
          pluginRenderers={DocViewerRenderers}
          documents={docs}
          className="my-doc-viewer-style"
          style={{width: 1350, height: 500}} 
          theme={{
            primary: "#5296d8",
            secondary: "#ffffff",
            tertiary: "#5296d899",
            textPrimary: "#ffffff",
            textSecondary: "#5296d8",
            textTertiary: "#00000099",
            disableThemeScrollbar: false,
          }}
          prefetchMethod="GET"
          config={{
            header: {
              disableHeader: false,
              disableFileName: false,
              retainURLParams: false,
            },
          }}
        />
      )}
      
    </div>
  );
};

export default App;
