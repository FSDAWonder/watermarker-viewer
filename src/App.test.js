import { render, screen } from '@testing-library/react';
import App from './App';
import DocViewer from "@cyntler/react-doc-viewer";

test('component render or not', () => {
  render(<App />);
  const linkElement = screen.getByText(/Document Viewer/i);
  expect(linkElement).toBeInTheDocument();
});

test("component render with no documents", () => {
  render(<DocViewer documents={[]} />);
  const linkElement = screen.getByTestId("react-doc-viewer");
  expect(linkElement).toMatchSnapshot();
});
