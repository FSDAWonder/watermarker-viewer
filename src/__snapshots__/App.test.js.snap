// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component render with no documents 1`] = `
<div
  class="sc-dtBdUo iWaeFc"
  data-testid="react-doc-viewer"
  id="react-doc-viewer"
>
  <div
    class="sc-imWYAI dTuydp"
    data-testid="header-bar"
    id="header-bar"
  />
  <div
    class="sc-jXbUNg hysiap"
    data-testid="proxy-renderer"
    id="proxy-renderer"
  >
    <div
      id="no-documents"
    />
  </div>
</div>
`;

exports[`renders component with no documents 1`] = `
<div
  class="sc-dtBdUo iWaeFc"
  data-testid="react-doc-viewer"
  id="react-doc-viewer"
>
  <div
    class="sc-imWYAI dTuydp"
    data-testid="header-bar"
    id="header-bar"
  />
  <div
    class="sc-jXbUNg hysiap"
    data-testid="proxy-renderer"
    id="proxy-renderer"
  >
    <div
      id="no-documents"
    />
  </div>
</div>
`;
